### PR TITLE
Added sphinx documentation for data shifts routine.

### PR DESCRIPTION
**Pull request by kperrynrel**
## Description

*Thank you for your contribution! Please provide a brief description of the problem and the proposed solution or new feature (if not already fully described in a linked issue)*

## Checklist

This PR is dependent on PR https://github.com/pvlib/pvanalytics/pull/124. It contains the Sphinx documentation for the data shift algorithm.

The following items must be addressed before the code can be merged. 
Please don't hesitate to ask for help if you are unsure of how to accomplish any of the items. 
*You are free to remove any checklist items that do not apply or add additional items that are 
not on this list*

- [X] Clearly documented all new API functions with [PEP257](https://www.python.org/dev/peps/pep-0257/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings
- [X] Adds description and name entries in the appropriate "what's new" file 
      in [`docs/whatsnew`](https://github.com/pvlib/pvanalytics/tree/master/docs/whatsnew) 
      for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` 
      or this Pull Request with `` :pull:`num` ``. Includes contributor name 
      and/or GitHub username (link with `` :ghuser:`user` ``).
- [X] Non-API functions clearly documented with docstrings or comments as necessary
- [X] Pull request is nearly complete and ready for detailed review

### CODE REVIEW

File: docs/examples/data-shifts.py
Comment (kanderso-nrel) [inline docs/examples/data-shifts.py:3]:
```suggestion
Data Shift Detection & Removal
==============================
```

File: docs/examples/data-shifts.py
Comment (kanderso-nrel) [inline docs/examples/data-shifts.py:5]: It's always nice to use sphinx's cross-referencing syntax (``` :py:mod:`pvanalytics.quality.data_shifts` ```) when referring to a specific part of the package.  In this case it won't actually link anywhere because we don't have a specific definition of the `data_shifts` module, but it'll still format it as code instead of plaintext.

File: docs/examples/data-shifts.py
Comment (kanderso-nrel) [inline docs/examples/data-shifts.py:83]: This plot seems to include a tiny bit of the pre-shift data, which is a bit unfortunate.  Should we add a bit of buffer to this example to prevent that?  Or maybe tweak the underlying function somehow?

image

**Review by kanderso-nrel — COMMENTED** [review]
The docs build is going to fail here until #124 is merged, so I think we should wait on doing a real review until then, but I built it locally and it looks pretty good!  Here are some comments in the meantime.

File: docs/examples/data-shifts.py
Comment (kperrynrel) [reply docs/examples/data-shifts.py:83]: We could run outlier removal, so it'd get any bad data out, or include a 5 day buffer in the filter. Will look into what I can do here today.

Comment (kperrynrel) [reply docs/examples/data-shifts.py:83]: Ok, got this issue addressed in commit https://github.com/pvlib/pvanalytics/pull/124/commits/3a29b8d3a841464f09f9a7e19f5e50850677c8b1. I added a 7-day buffer to the start end dates of the longest period to filter anything out from previous or future data shift periods. Here's what the new graphic for this example looks like:
example_1_filtered

File: docs/whatsnew/0.1.1.rst
Comment (kanderso-nrel) [inline docs/whatsnew/0.1.1.rst:41]: We should be sure to move this to 0.1.2.rst before merging

**Comment by kanderso-nrel** [comment]
@kperrynrel I think the docs build failure will be resolved by adding a new item `- optional` under `doc` in this bullet list in the RTD config:

https://github.com/pvlib/pvanalytics/blob/e8ce80154eecabf9088e8c551bd7ebebb79b2e32/.readthedocs.yml#L13-L15

**Comment by kperrynrel** [comment]
@kanderso-nrel thanks for the tip on the readthedocs.yml--that would've taken me hours to debug!!

File: docs/examples/data-shifts.py
Comment (kperrynrel) [reply docs/examples/data-shifts.py]: Changed to ds in https://github.com/pvlib/pvanalytics/pull/131/commits/00cef96cefb6c0e8f43660f8806dea5154db6b45

File: docs/examples/data-shifts.py
Comment (kperrynrel) [reply docs/examples/data-shifts.py]: Chnaged to pvlib in https://github.com/pvlib/pvanalytics/pull/131/commits/00cef96cefb6c0e8f43660f8806dea5154db6b45

File: docs/examples/data-shifts.py
Comment (kperrynrel) [reply docs/examples/data-shifts.py:54]: @kanderso-nrel  I just updated the plots in the documentation based on your suggestions--let me know what you think!

**Review by kanderso-nrel — APPROVED** [review]
LGTM, thanks @kperrynrel!

**Review by cwhanse — APPROVED** [review]
Sorry for the delay @kperrynrel

**Pull request merged**
